### ISSUE TITLE
Adapt ActuatorMetricsEliminator and BackgroundPreInitializerDeactivator to build-time transformations

### ISSUE THREAD

**Issue by sdeleuze** [spring-projects-experimental/spring-native#472]

**Comment by aclement** [comment]
@sdeleuze @bclozel - any thoughts on this:

The behaviour of `BackgroundPreInitializerDeactivator` in the feature is already implemented for bti - but in a rather harsh way perhaps. In the `DefaultFactoriesCodeContributor` that produces code for `StaticSpringFactories` it is filtered out. This effectively means using the BTI it doesn't give you an optimal fat jar JVM experience (because there is no backgroundpreinitializer included). Instead it gives you a fat jar that works best when compiled to native image.  In this situation it feels a bit more like we want to generate `StaticSpringFactories` that looks like this kind of pattern:

```
factories.register(...);
factories.register(...);
factories.register(...);
if (!IN_NATIVE_IMAGE) {
  factories.register(....BackgroundPreInitializer::new);
}
factories.register(...);
factories.register(...);
```

That would give us the best of both worlds - an optimal fat jar that can benefit from bti and use the pre initializer, and then a native-image that would exclude it. Generating a simple condition like that would seem doable. I suppose it depends on what you think the aim of BTI is - is it just a route to optimal native-image or is it also a route to an optimal fat jar too. My preference is the latter. (If we had more than one candidate I'd much prefer an annotation on the types that indicate we don't need them in a native image - almost a `@ConditionalOnNotNativeImage` or something, but of course this isn't configuration, it's for anything you could list in spring.factories.

**Comment by aclement** [comment]
With the shift of all the relevant feature code into the `buildtools` module, it actually becomes relative straightforward to utilize the exact same `ActuatorMetricsEliminator` and `BackgroundPreInitializerDeactivator`. They aren't particularly messy because they don't use the type system so aren't caught between old and new type system variants. So I've turned them on. (This works, but doesn't address my previous comment).

Coupled with now having build time support for property evaluation, the `actuator-webmvc` sample changes from
```
Build memory: 6.48GB
Image build time: 124.9s
RSS memory: 78.4M
Image size: 64.3M
Startup time: 0.133 (JVM running for 0.134)
```
to now:
```
Build memory: 6.52GB
Image build time: 121.9s
RSS memory: 64.9M (13.5M saved)
Image size: 62.8M
Startup time: 0.102 (JVM running for 0.103)
```

As a reminder, with the feature it was:
```
Build memory: 6.71GB
Image build time: 118.2s
RSS memory: 65.4M
Image size: 59.8M
Startup time: 0.105 (JVM running for 0.107)
```

So bti is extra saving of 0.5M RSS over the feature.

I modified the `compileWithMaven.sh` script so that it passes through `-D` options specified in the build.sh scripts in the samples. The actuator samples are turning on the actuator metrics eliminator and property processing.
